apps/dashboard/lib/trpc/routers/workspace/onboarding.ts

```diff
@@ -0,0 +1,93 @@
+import { createKeyInputSchema } from "@/app/(app)/apis/[apiId]/_components/create-key/create-key.schema";
+import { db } from "@/lib/db";
+import { TRPCError } from "@trpc/server";
+import { z } from "zod";
+import { requireUser, t } from "../../trpc";
+import { createApiCore } from "../api/create";
+import { createKeyCore } from "../key/create";
+
+const createWorkspaceWithApiAndKeyInputSchema = z.object({
+  workspaceId: z.string(),
+  apiName: z
+    .string()
+    .min(3, "API name must be at least 3 characters")
+    .max(50, "API name must not exceed 50 characters"),
+  ...createKeyInputSchema.omit({ keyAuthId: true }).shape,
+});
+
+export const onboardingKeyCreation = t.procedure
+  .use(requireUser)
+  .input(createWorkspaceWithApiAndKeyInputSchema)
+  .mutation(async ({ input, ctx }) => {
+    const { workspaceId, apiName, ...keyInput } = input;
```

Comment:
never ever ever use a workspace id from input, we've done that in the past and it's a security issue, use ctx.workspace.id